### PR TITLE
TLS allowed CommonNames

### PR DESCRIPTION
**Pull request by kmlebedev**
https://github.com/chrislusf/seaweedfs/issues/1841
https://jbrandhorst.com/post/grpc-auth/

### CODE REVIEW

File: docker/compose/tls.env [hunk 1]
```diff
@@ -0,0 +1,13 @@
+WEED_GRPC_CA=/etc/seaweedfs/tls/SeaweedFS_CA.crt
+WEED_GRPC_MASTER_CERT=/etc/seaweedfs/tls/master01.crt
+WEED_GRPC_MASTER_KEY=/etc/seaweedfs/tls/master01.key
+WEED_GRPC_VOLUME_CERT=/etc/seaweedfs/tls/volume01.crt
+WEED_GRPC_VOLUME_KEY=/etc/seaweedfs/tls/volume01.key
+WEED_GRPC_FILER_CERT=/etc/seaweedfs/tls/filer01.crt
+WEED_GRPC_FILER_KEY=/etc/seaweedfs/tls/filer01.key
+WEED_GRPC_CLIENT_CERT=/etc/seaweedfs/tls/client01.crt
+WEED_GRPC_CLIENT_KEY=/etc/seaweedfs/tls/client01.key
+WEED_GRPC_MASTER_ALLOWED_COMMONNAMES="volume01 master01 filer01 client01"
```
Comment (chrislusf) [inline docker/compose/tls.env]: is this variable a string or list or comma separated list?

Comment (kmlebedev) [reply docker/compose/tls.env]: > is this variable a string or list or comma separated list?
This is space separated list

File: weed/command/scaffold.go [hunk 2]
```diff
@@ -444,26 +444,29 @@ ca = ""
 [grpc.volume]
 cert = ""
 key  = ""
+allowed_commonNames = ""
```
Comment (chrislusf) [inline weed/command/scaffold.go]: need to add comments that these are separated by empty space

File: docker/compose/tls.env (hunk 1 above)
Comment (chrislusf) [reply docker/compose/tls.env]: actually, can this be changed to comma separated instead?

Comment (kmlebedev) [reply docker/compose/tls.env]: > actually, can this be changed to comma separated instead?

I have reused the function config.GetStringSlice
https://github.com/chrislusf/seaweedfs/blob/52a8f1470ef92b9d0075c3dccc76a4170ab30c07/weed/util/config.go#L83
You can expand it or declare a new one, but I could not do it on the fly

**Comment by kmlebedev** [comment]
@chrislusf  changed to commas and added a comment

**Pull request merged**